lib/core/src/Cardano/Wallet/Primitive/Types/Tx.hs

```diff
@@ -620,23 +651,8 @@ cardanoTxFromBytes
     -> ByteString -- ^ Serialised transaction
     -> Either DecoderError (InAnyCardanoEra Cardano.Tx)
 cardanoTxFromBytes maxEra bs = asum $ map snd $ filter (withinEra maxEra . fst)
-    -- NOTE: Attempting to deserialise in Alonzo before Babbage is intentional.
-    --
-    -- It seems Alonzo transactions can be deserialized as Babbage
-    -- transactions. This causes new-tx workflow integration tests to fail.
-    -- The txs are interpreted as Babbage txs right before submitting them to
-    -- the Alonzo-era ledger.
-    --
-    -- This may also cause problems with our store of pending txs.
```

Comment:
Right! 😬 This might be a very real and relevant problem https://github.com/input-output-hk/cardano-wallet/blob/68d1bc9c5450f0400794c30b1cefee1c2f35a93e/lib/core/src/Cardano/Wallet.hs#L2661

which goes silently un-noticed with the PR as is.

Not sure why I didn't try _changing_ the implementation of `SealedTx` rather than replacing it in the API. Will try that next.